tests/ecmascript/test-bi-object-assign.js

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Object.assign() (E7 19.1.2.1)
+ *  http://www.ecma-international.org/ecma-262/7.0/#sec-object.assign
+ */
+
+/*===
+basic tests
+true
+maggie Kittycow Machel Fatty Whale 812 1208 9001
+0,1,2,pig,cow,ape,whale
+maggie 1208 9001 Vegeta
+cow,pig,ape,saiyan
+undefined
+this belongs here
+undefined
+prop1
+get
+set 812
+===*/
+
+function basicTest() {
+    var obj, src, dest;
+
+    // If no coercion is needed, target object is returned (i.e. not a copy).
+    obj = {};
+    dest = Object.assign(obj, { test: "test" });
+    print(dest === obj);
+
+    // Properties are assigned in [[OwnPropertyKeys]] order, moving from left
+    // to right in the argument list:
+    //     1. Array indices (sequential, ascending)
+    //     2. String keys (creation order)
+    //     3. Symbol keys (creation order)
+    // This behavior is easy to test for by assigning from multiple sources to
+    // an empty object and then calling Reflect.ownKeys() on the result.
```

Comment:
So just to be clear:

> Reflect.ownKeys() is specified to list the keys in insertion order, so checking that order is (again, from black-box perspective) a good indicator that Object.assign() did the right thing.

My point here was that for array index inserts Reflect.ownKeys() actually won't list keys in the insertion order. So assuming Reflect.ownKeys() works correctly but Object.assign() handles array indices in an incorrect order, Reflect.ownKeys() (if working correctly) will effectively mask the error.

Reflect.ownKeys() is good for checking that non-array-index keys were handled correctly, agree with that.